app/api/v_sphere/virtual_machine.rb

```diff
@@ -111,11 +111,15 @@ def reset_vm
     end
 
     def shutdown_guest_os
-      @vm.ShutdownGuest.wait_for_completion if powered_on?
+      # Do not `wait_for_completion` here, as `ShutdownGuest` does not offer a useful return value
+      # https://code.vmware.com/apis/196/vsphere#/doc/vim.VirtualMachine.html#shutdownGuest
+      @vm.ShutdownGuest if powered_on?
     end
 
     def reboot_guest_os
-      @vm.RebootGuest.wait_for_completion
+      # Do not `wait_for_completion` here, as `RebootGuest` does not offer a useful return value
+      # https://code.vmware.com/apis/196/vsphere#/doc/vim.VirtualMachine.html#rebootGuest
+      @vm.RebootGuest
```

Comment:
Does this need `if powered_on?`? Can a guest be rebooted if it is powered off?

Reply:
No, it can't. See:
![screenshot from 2019-03-05 12-43-03](https://user-images.githubusercontent.com/24682367/53803197-46d1ee80-3f44-11e9-9dc8-dd3a88ee4a7c.png)

Reply:
Thanks for the info! Should the `if powered_on?` condition be added here then?

Reply:
@BraunTom Is there a reason it isn't here?

Reply:
@chrisma Fixed in https://github.com/hpi-swt2/vm-portal/pull/442/commits/6601a7324719579d0107cf2009e40f04840d5f8a.